restricted_site_access.php

```diff
@@ -216,35 +216,25 @@ public static function restrict_access( $wp ) {
 
 		// check for the allow list, if its empty block everything
 		if ( ! empty( self::$rsa_options['allowed'] ) && is_array( self::$rsa_options['allowed'] ) ) {
-			$remote_ip = $_SERVER['REMOTE_ADDR'];  // save the remote ip
-			if ( strpos( $remote_ip, '.' ) ) {
-				$remote_ip = str_replace( '::ffff:', '', $remote_ip ); // handle dual-stack addresses
-			}
-			$remote_ip = inet_pton( $remote_ip ); // parse the remote ip
+			$remote_ip = get_client_ip_address(); //save the remote ip
 
 			// iterate through the allow list
-			foreach ( self::$rsa_options['allowed'] as $line ) {
-				list( $ip, $mask ) = explode( '/', $line . '/128' ); // get the ip and mask from the list
-
-				$mask = str_repeat( 'f', $mask >> 2 ); // render the mask as bits, similar to info on the php.net man page discussion for inet_pton
-
-				switch ( $mask % 4 ) {
-					case 1:
-						$mask .= '8';
-						break;
-					case 2:
-						$mask .= 'c';
-						break;
-					case 3:
-						$mask .= 'e';
-						break;
-				}
-
-				$mask = pack( 'H*', $mask );
-
-				// check if the masked versions match
-				if ( ( inet_pton( $ip ) & $mask ) == ( $remote_ip & $mask ) ) {
-					do_action( 'restrict_site_access_ip_match', $remote_ip, $ip, $mask ); // allow users to hook ip match
+			foreach( self::$rsa_options['allowed'] as $line ) {
+				if( ip_in_range( $remote_ip, $line ) ){
+					do_action( 'restrict_site_access_ip_match', $remote_ip ); // allow users to hook ip match
+
+					/**
+					 * Fires when an ip address match occurs.
+					 *
+					 * Enables adding session_start() to the IP check, ensuring Varnish type cache will not cache the request.
+					 *
+					 * @since 6.0.2
+					 *
+					 * @param string $remote_ip The remote IP address being checked.
+					 * @param string $ip        The matched IP address.
+					 * @param string $mast      The IP mask used in the match.
```

Comment:
wound up removing this line because we no longer have ip/mask here. since this is a brand new action, feels fine to change this and just pass what we have - namely the entire matched line